packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -111,49 +122,49 @@ export class CDKDeployer implements BackendDeployer {
     deploymentType?: DeploymentType,
     additionalArguments?: string[]
   ): Promise<DeployResult | DestroyResult> => {
-    // Basic args
-    const cdkCommandArgs = [
-      'cdk',
-      invokableCommand.toString(),
-      // This is unfortunate. CDK writes everything to stderr without `--ci` flag and we need to differentiate between the two.
-      // See https://github.com/aws/aws-cdk/issues/7717 for more details.
-      '--ci',
-      '--app',
-      `'npx tsx ${this.backendLocator.locate()}'`,
-      '--all',
-      '--output',
-      '.amplify/artifacts/cdk.out',
-    ];
-
-    // Add context information if available
-    if (backendId) {
-      cdkCommandArgs.push(
-        '--context',
-        `${CDKContextKey.BACKEND_NAMESPACE}=${backendId.namespace}`,
-        '--context',
-        `${CDKContextKey.BACKEND_NAME}=${backendId.name}`
-      );
+    try {
```

Comment:
rather than nest this whole method in a try catch, I'd rather create a `private tryInvokeCdk()` method that calls this one within a try/catch and then use that method elsewhere in the class for invoking cdk

Reply:
updated